connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExactlyOnceSourceIntegrationTest.java

```diff
@@ -826,15 +830,22 @@ public void testSeparateOffsetsTopic() throws Exception {
             assertConnectorStopped(connectorStop);
 
             // consume all records from the source topic or fail, to ensure that they were correctly produced
-            ConsumerRecords<byte[], byte[]> records = connectorTargetedCluster.consumeAll(
+            ConsumerRecords<byte[], byte[]> sourceRecords = connectorTargetedCluster.consumeAll(
                     CONSUME_RECORDS_TIMEOUT_MS,
                     Collections.singletonMap(ConsumerConfig.ISOLATION_LEVEL_CONFIG, "read_committed"),
                     null,
                     topic
             );
-            assertTrue("Not enough records produced by source connector. Expected at least: " + recordsProduced + " + but got " + records.count(),
-                    records.count() >= recordsProduced);
-            assertExactlyOnceSeqnos(records, numTasks);
+            assertTrue("Not enough records produced by source connector. Expected at least: " + recordsProduced + " + but got " + sourceRecords.count(),
+                    sourceRecords.count() >= recordsProduced);
+            // also have to check which offsets have actually been committed, since we no longer have exactly-once guarantees
```

Comment:
I don't quite follow the "no longer have EOS guarantees" part. And that made me notice that this test method (unlike the others in the class) doesn't have a javadoc description about what exactly is being tested here.

Reply:
Good point; added a Javadoc.

I'll note that, while writing it, it's occurred to me that we could strengthen some of the guarantees in the test. For example, we don't assert that the connector has actually committed any new offsets after bringing up the cluster again with exactly-once support disabled. But I'm a little reluctant to address these immediately since it may introduce additional flakiness if we get things wrong, and this PR is likely going to be backported to 3.3 in order to improve stability while generating release candidates. If you believe these additional guarantees are worth pursuing, though, let me know and I can file a separate follow-up knowing that it's targeted exclusively at trunk.